File: react-java/backend/src/main/java/com/clt/hrm/infra/common/dtos/SortDto.java
package com.clt.hrm.infra.common.dtos;

import jakarta.validation.constraints.Pattern;
import lombok.Data;
import com.fasterxml.jackson.annotation.JsonPropertyDescription;

@Data
public class SortDto {
	@JsonPropertyDescription("Sort field. Search by sort field.")
	@Pattern(regexp = "^[A-Za-z0-9_]+$", message = "Invalid SortField")
	private String sortField;

	@JsonPropertyDescription("Sort type. Search by sort type. Values: ASC (Ascending), DESC (Descending)")
	@Pattern(regexp = "ASC|DESC", message = "SortType must be ASC or DESC")
	private String sortType;
}

File: nestjs/backend/src/infra/common/dto/sort.dto.ts
import { IsIn, IsOptional, Matches } from 'class-validator';

export class SortDto {
  @IsOptional()
  @Matches(/^[A-Za-z0-9_]+$/, { message: 'sortField must be alphanumeric with underscores only' })
  sortField?: string;

  @IsOptional()
  @IsIn(['ASC', 'DESC'])
  sortType?: 'ASC' | 'DESC';
}
